R/get_predicted_args.R

```diff
@@ -0,0 +1,217 @@
+      warning(msg, call. = FALSE)
+    }
+  }
+
+  # Get info
+  info <- model_info(x, verbose = FALSE)
+
+  # Data
+  if (!is.null(newdata) && is.null(data)) data <- newdata
+  if (is.null(data)) data <- get_data(x, verbose = verbose)
+
+  # CI
+  if (is.null(ci)) ci <- 0
+
+  # check `predict` user-input
+  predict_method <- lapply(
+    class(x), function(i) {
+      tryCatch(utils::getS3method("predict", i),
+               error = function(e) NULL)
+    }
+  )
+  predict_method <- tryCatch(predict_method[!sapply(predict_method, is.null)][[1]], error = function(e) NULL)
+
+  # check aliases
+  if (predict %in% c("expected", "response")) {
+    predict <- "expectation"
+  }
+  if (predict == "predicted") {
+    predict <- "prediction"
+  }
+
+  ## TODO remove in a later update
+  # backward compatibility
+  if (identical(predict, "relation")) {
+    message(format_message(
+      '`predict = "relation" is deprecated.',
+      'Please use `predict = "expectation" instead.'
+    ))
+    predict <- "expectation"
+  }
+
+  # Warn if get_predicted() is not called with an easystats- or
+  # model-supported predicted type
+  easystats_methods <- c("expectation", "link", "prediction", "classification")
+  type_methods <- suppressWarnings(eval(formals(predict_method)$type))
+  supported <- c(easystats_methods, type_methods)
+  if (isTRUE(verbose) && !is.null(predict) && !predict %in% supported) {
+    msg <- format_message(
+      sprintf('`predict` = "%s"` is not officially supported by `get_predicted()`.', predict),
+      '`predict` will be passed directly to the `predict()` method for the model and not validated.',
+      'Please check the validity and scale of the results.',
+      'Set `verbose = FALSE` to silence this warning, or use one of the supported values for the `predict` argument:',
+      paste(" ", paste(sprintf('"%s"', setdiff(easystats_methods, c("expected", "predicted"))), collapse = ", "))
+    )
+    warning(msg, call. = FALSE)
+  }
+
+
+  transform <- FALSE
+
+  # type_arg  = what we pass own to type
+  # scale_arg = the scale of the predictions (link, response, terms, ...)
+  #             this may differ from type_arg, because we back-transform
+  # transform = whether we back-transform predictions, SE and CI
+
+  # define prediction type (that's for the initial call to stats::predict)
+
+  # terms
+  if (predict == "terms") {
+    type_arg <- "terms"
+    scale_arg <- "terms"
+    transform <- FALSE
+
+    # linear models are always on response scale (there is no other)
+  } else if (info$is_linear) {
+    type_arg <- "response"
+    scale_arg <- "response"
+    transform <- FALSE
+
+    # type = "response" always on link-scale - for later back-transformation
+  } else if (predict %in% c("expectation", "response", "prediction", "classification")) {
+
+    if (inherits(x, c("hurdle", "zeroinfl", "zerotrunc"))) {
+      # pscl: hurdle/zeroinfl and countreg
+      type_arg <- "count"
```

Comment:
Should again warn or error. Maybe use `match.arg`

Reply:
What do you exactly mean here?

Reply:
I can't tell from this chunk of code, but I just wanted to make sure that we warn users whenever the`type` is changed on the fly automatically. If we can always transform it back to the scale the user originally requested, there is no need to include a warning and we can move on. But if we force it to `count` and then convert it when we can -- leaving some unknown cases out -- then we need a warning.